chrome/content/preferences/prefs-ce.js

```diff
@@ -0,0 +1,1784 @@
+    };
+  }
+
+  static get inheritedAttributes() {
+    return Object.assign({}, super.inheritedAttributes,
+        PaneButton.thisInheritedAttributes);
+  }
+
+  /**
+   * Overrides super.getElementForAttrInheritance. Used in setting up attribute inheritance.
+   * Takes a selector and returns an element for that selector. If the selector is in
+   * thisInheritedAttributes it queries the light DOM, otherwise the selector is in
+   * super.inheritedAttributes and the shadow DOM is queried. See bug 1545824.
+   *
+   * @param {string} selector  A selector used to query an element.
+   * @return {Element}  The element found by the selector.
+   */
+  getElementForAttrInheritance(selector) {
+    if (selector in PaneButton.thisInheritedAttributes) {
+      return this.querySelector(selector);
+    }
+    return super.getElementForAttrInheritance(selector);
+  }
+
+  get fragment() {
+    if (!this._fragment) {
+      this._fragment = MozXULElement.parseXULToFragment(`
+          <image class="paneButtonIcon"/>
+          <label class="paneButtonLabel"/>
+      `);
+    }
+    return this.ownerDocument.importNode(this._fragment, true);
+  }
+
+  connectedCallback() {
+    if (this._initialized) {
+      return;
+    }
+
+    super.connectedCallback();
+    this.appendChild(this.fragment);
+    this.initializeAttributeInheritance();
+    this._initialized = true;
+  }
+
+  get disabled() {
+    return this.getAttribute('disabled') == 'true';
+  }
+
+  set disabled(val) {
+    if (val)
+      this.setAttribute('disabled', 'true');
+    else
+      this.removeAttribute('disabled');
+    return val;
+  }
+
+  get tabIndex() {
+    return parseInt(this.getAttribute('tabindex')) || 0;
+  }
+
+  set tabIndex(val) {
+    if (val)
+      this.setAttribute('tabindex', val);
+    else
+      this.removeAttribute('tabindex');
+    return val;
+  }
+
+  get label() {
+    return this.getAttribute('label');
+  }
+
+  set label(val) {
+    this.setAttribute('label',val);
+    return val;
+  }
+
+  get crop() {
+    return this.getAttribute('crop');
+  }
+
+  set crop(val) {
+    this.setAttribute('crop',val);
+    return val;
+  }
+
+  get image() {
+    return this.getAttribute('image');
+  }
+
+  set image(val) {
+    this.setAttribute('image',val);
+    return val;
+  }
+
+  get command() {
+    return this.getAttribute('command');
+  }
+
+  set command(val) {
+    this.setAttribute('command',val);
+    return val;
+  }
+
+  get accessKey() {
+    return this.getAttribute("accesskey");
+  }
+
+  set accessKey(val) {
+    // Always store on the control
+    this.setAttribute("accesskey", val);
+    // If there is a label, change the accesskey on the labelElement
+    // if it's also set there
+    let labelElement = document.getElementsByAttribute("control", this.id)[0];
+    if (labelElement) {
+      labelElement.setAttribute("accesskey", val);
+    }
+    return val;
+  }
+}
+
+class PrefWindow extends MozXULElement {
```

Comment:
Where PrefWindow is use ?

Reply:
The option dialog window of the add-on

Reply:
The definition of the prefwindow CE is in the .xhtml file at the onload attr of the dialog window for the reason that some elements have to had thier inner code placed before the object is instanced.

Reply:
[link](https://github.com/117649/TabMixPlus/blob/d9586368ebf1f33b06c5f857a5849563d72d1eef/chrome/content/preferences/preferences.xhtml#L23)